classes/order/Order.php

```diff
@@ -1673,9 +1673,12 @@ public function getWsOrderRows()
      */
     public function setCurrentState($id_order_state, $id_employee = 0)
     {
-        if (empty($id_order_state)) {
+        if (empty($id_order_state) || $id_order_state == $this->current_state) {
             return false;
         }
+        if ($id_order_state == $this->current_state) {
```

Comment:
unnecessary 🤔 ?

Reply:
Yes, you are right, a mistake pasting the code, let me fix it.